### ISSUE TITLE
[Issue]: 6.3.0 source archive was modified after release

### ISSUE THREAD

**Issue by AngryLoki** [ROCm/Tensile#2067]
### Problem Description

Hi, 

[Our user](https://github.com/gentoo/gentoo/pull/39654#issuecomment-2545532632) detected that `rocm-6.3.0.tar.gz` archive changed the checksum. 

On further investigation comparing to https://github.com/ROCm/Tensile/commits/rocm-6.3.0 I see that 2 different archives https://github.com/ROCmSoftwarePlatform/Tensile/archive/rocm-6.3.0.tar.gz were released:
* On Dec  7 downloaded archive had size 12724464 bytes
* On Dec 17 downloaded archive has size 12724004 bytes and contains 3 extra commits https://github.com/ROCm/Tensile/commit/5838cf77c4554fcdd18fe633027f4c5d69c2f656 https://github.com/ROCm/Tensile/commit/000e50346a9d8e3f56631d09ecac916dad4bf068 https://github.com/ROCm/Tensile/commit/aca95d1743c243dd0dd0c8b924608bc915ce1ae7

Since source code archives are checksummed, distributed across mirrors of many distributions, and provide specific functionality (which was changed in these 3 commits), could you please release a patch release (with a common release procedure, so that maintainers are notified about these changes) next time instead of reattaching an existing tag to a different commit? Thanks!

### Operating System

Gentoo

### CPU

-

### GPU

-

### ROCm Version

ROCm 6.3.0

**Comment by ppanchad-amd** [comment]
Hi @AngryLoki. Internal ticket has been created to assist with your issue. Thanks!

**Comment by bstefanuk** [comment]
@AngryLoki Thanks for your issue. I am the one who promoted those commits to the ROCm 6.3.0 release and want to confirm that this is very much a one-off update that we had to weigh the trade-offs for. In this case, those three commits were included downstream in the 6.3.0 release of rocBLAS, so we had to reconcile the releases. Since patch releases are also cadenced in the ROCm stack, I couldn't just issue a patch release. This type of update is highly atypical and wasn't made without consideration, we apologize for any impact it had on your users. Going forward, our staging and release branches will be updated properly and you shouldn't see this type of problem again.
